qemu/tests/seabios_hotplug_unplug.py

```diff
@@ -0,0 +1,177 @@
+import re
+import logging
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import utils_net
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    KVM Seabios test:
+    1) Start guest with sga bios
+    2) Check the sga bios messages(optional)
+    3) Hotplug a virtio-blk-pci device and a virtio-net-pci device
+    4) Restart the guest
+    5) Check whether the hotplugged devices are in the boot menu
+    6) Hotunplug the hotplugged virtio-blk-pci and virtio-net-pci devices
+    7) Restart the guest
+    8) Check whether the hotunplugged devices are still in the boot menu
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_output(session_obj):
+        """
+        Use the function to short the lines in the scripts
+        """
+        return session_obj.get_stripped_output()
+
+    def sga_info_check():
+        return re.search(sgabios_info, get_output(vm.serial_console))
+
+    def menu_hint_check():
+        return (len(re.findall(boot_menu_hint,
+                               get_output(seabios_session))) > reboot_times)
+
+    def get_list():
+        return re.findall(r"^\d+\. (.*)\s", get_output(seabios_session), re.M)
+
+    def get_boot_menu():
+        if not utils_misc.wait_for(menu_hint_check, timeout, 1):
+            test.fail("Could not get boot menu message")
+        vm.send_key(boot_menu_key)
+        boot_list_after = utils_misc.wait_for(get_list, timeout, 1)
+        menu_list = boot_list_after[len(boot_list_before):]
```

Comment:
So, do you intent to use `menu_list` to get the differential of devices between two times of booting? If yes, there would be two issues against this approach:

1. there is no guarantee that newly added devices always be appended to the list, it depends on the `bootindex` of the device, for example, you may see:
```
# first boot
1. a
2. b

# second boot
1. a
2. c
3. b
```
then you will get `b` in the `menu_list` which is unexpected, however, in this case it won't happen since you haven't specified `bootindex`, so it would be fine.

2. by hot-unplugging, `menu_list` will always be empty no matter how many devices were disappeared actually, therefore, you can't get correct information to verify the function.

Reply:
This is meant to get the latest boot menu list.
And I have added a new function get_diff() to get the differential of devices between two times of booting in the latest code.
Thanks!